deep_field_metadetect/gaussmom/gaussmom.py

```diff
@@ -0,0 +1,417 @@
+
+    pars = jnp.array(
+        [
+            sums[mv_ind],
+            sums[mu_ind],
+            res_e1,
+            res_e2,
+            res_T,
+            res_flux,
+        ]
+    )
+
+    res_flagstr = ngmix.flags.get_flags_str(res_flags)
+    res_flux_flagstr = ngmix.flags.get_flags_str(res_flux_flags)
+    res_T_flagstr = ngmix.flags.get_flags_str(res_T_flags)
+
+    res = GaussMomData(
+        obs=obs,
+        flags=res_flags,
+        flagstr=res_flagstr,
+        wsum=sums_norm,
+        flux=res_flux,
+        sums=res_sums,
+        sums_cov=res_sums_cov,
+        sums_norm=res_sums_norm,
+        flux_flags=res_flux_flags,
+        flux_flagstr=res_flux_flagstr,
+        T_flags=res_T_flags,
+        T_flagstr=res_T_flagstr,
+        flux_err=res_flux_err,
+        T=res_T,
+        T_err=res_T_err,
+        s2n=res_s2n,
+        e1=res_e1,
+        e2=res_e2,
+        e=res_e,
+        e_err=res_e_err,
+        e_cov=res_e_cov,
+        sums_err=res_sums_err,
+        pars=pars,
+    )
+
+    # _add_moments_by_name(res) # shouldn't modify sums or sums_cov
+
+    return res
+
+
+def _add_moments_by_name(res):  # TODO
+    sums = res["sums"]
+    sums_cov = res["sums_cov"]
+
+    mf_ind = MOMENTS_NAME_MAP["MF"]
+    fsum = sums[mf_ind]
+    fsum_err = np.sqrt(sums_cov[mf_ind, mf_ind])
+
+    # add in named sums normalized by flux sum (weight * image).sum()
+    # we don't store flags or errors for these
+    mkeys = list(MOMENTS_NAME_MAP.keys())
+    for name in mkeys:
+        print(name)
+        ind = MOMENTS_NAME_MAP[name]
+        if ind > sums.size - 1:
+            continue
+
+        err_name = f"{name}_err"
+
+        if name in ["MF", "M00"]:
+            res[name] = fsum
+            res[err_name] = fsum_err
+        else:
+            if fsum > 0:
+                res[name] = sums[ind] / fsum
+                res[err_name] = get_ratio_error(
+                    sums[ind],
+                    sums[mf_ind],
+                    sums_cov[ind, ind],
+                    sums_cov[mf_ind, mf_ind],
+                    sums_cov[ind, mf_ind],
+                )
+            else:
+                res[name] = np.nan
+                res[err_name] = np.nan
+
+
+@jax.jit
+def get_ratio_error(a, b, var_a, var_b, cov_ab):
+    """
+    Compute the error on the ratio a / b using JAX.
+    """
+    var = get_ratio_var(a, b, var_a, var_b, cov_ab)
+
+    var = jnp.clip(var, 0.0, jnp.inf)
+    error = jnp.sqrt(var)
+    return error
+
+
+@jax.jit
+def get_ratio_var(a, b, var_a, var_b, cov_ab):
+    """
+    Compute the variance of (a/b).
+    """
+
+    # Ensure no division by zero
+    b = jnp.where(
+        b == 0, 1e-100, b
+    )  # TODO: This does not raise a value error like ngmix
+
+    rsq = (a / b) ** 2
+
+    var = rsq * (var_a / a**2 + var_b / b**2 - 2 * cov_ab / (a * b))
+    return var
+
+
+def eval_gaussian_moments(gaussmom_obs: GaussMomObs, fwhm: float, maxrad: float):
```

Comment:
I don't recall this function in the original ngmix API.

Reply:
I am making the API closer to ngmix now